### PR TITLE
[WIP] Add clear cache flags

### PR DESCRIPTION
**Pull request by kevinkassimo**
Added the following flags:
+ `-c` or `--clear-cache`: clear `.deno/gen` folder, remove cached compiled code
+ `-d` or `--clear-deps`: clear `.deno/deps` folder, remove cached remote resources

Also fix comments in `deno_dir.rs`

TODO: add tests to validate that they are working as intended?

### CODE REVIEW

**Comment by hayd** [comment]
I don't think this needs -c or -d.

Also, atm if you do `deno --clear-cache --clear-deps` it doesn't clear both or complain. Is there a reason this doesn't look more like `--refresh`, i.e. continue to run the script after clear.

File: src/flags.rs [hunk 1]
```diff
@@ -40,6 +50,26 @@ pub fn process(flags: &DenoFlags, usage_string: String) {
...
+      exit(0);
+    }
+  }
+
+  if flags.clear_deps {
+    if let Err(err) = deno_fs::remove_dir_contents_all(&state.dir.deps) {
+      eprintln!("Failed to clear dependencies: {}", err.description());
+      exit(1);
+    } else {
+      println!("Successfully cleared dependencies.");
```
Comment (hayd) [inline src/flags.rs]: maybe ought to be `debug!`?
I'd probably say `Successfully cleared dependencies: {}, &state.dir.deps)`

**Comment by kevinkassimo** [comment]
@hayd I can tweak the code to allow the two flags work at the same time.
As for refreshing behavior, we already have `--recompile` and thus I personally prefer not to allow run script immediately after this (like `deno --clear-cache --clear-deps script.ts` would not run the `script.ts`)

File: src/flags.rs (hunk 1 above)
Comment (kevinkassimo) [reply src/flags.rs]: `debug!` prints only when the user provides `-D`. I prefer to let the user know the effect of the flag.
Also I don't think `state.dir.deps` is an important info that should be exposed to the user.

Comment (hayd) [reply src/flags.rs]: They know the effect of the flag, they passed it!

**Comment by ry** [comment]
Why do we need this? Can't people just `rm -rf ~/.deno/deps` ?

**Comment by kevinkassimo** [comment]
@ry Are we going to explicitly document the cache folders? If yes then we probably don't need `--clear-*` (I don't think Go has such command)...

**Comment by kevinkassimo** [comment]
Closing this unless we have a strong case for these flags.

**Comment by ry** [comment]
@kevinkassimo yes, let's document the structure

**Pull request closed without merging**